### ISSUE TITLE
SQLiteConnectionPool issue

### ISSUE THREAD

**Issue by Dkhusainov** [requery/sqlite-android#64]
Hello. 

My code uses multiple thread to access database at a give timeframe. After couple of seconds inserts it stops giving access to db and outputs the following
`W/SQLiteConnectionPool: The connection pool for database '/data/user/0/group.infotech.reference.dev/databases/alternative-references.sqlite' has been unable to grant a connection to thread 1778 (RxCachedThreadScheduler-1) with flags 0x2 for 630.01605 seconds.
    Connections: 0 active, 1 idle, 2 available.`

Happens with both `writeAheadLogging`.

Any tips for me to figure out what exactly happening. Maybe some kind of deadlock?

**Comment by npurushe** [comment]
Hi there are a number of reasons this could happen, I would make sure if you are using transactions they are being ended correctly, and connection objects (cursors, statements) are closed properly when used from other threads.

**Comment by julianfalcionelli** [comment]
I faced with the same issue, I was doing I/O events on multiple threads.
I was using `Scheduler.io()` as subscribe thread and apparently that was causing some kind of deadlock so I switched the subscriptions to use `Schedulers.single()` to have them on background but synchronised. It helped a lot but after doing some stress testing (forcing a huge number of I/O events on same database) I get the same issue.
